src/main/java/org/opentripplanner/service/vehiclerental/street/Composite.java

```diff
@@ -0,0 +1,97 @@
+package org.opentripplanner.service.vehiclerental.street;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.opentripplanner.street.model.RentalRestrictionExtension;
+import org.opentripplanner.street.search.state.State;
+
+/**
+ * Combines multiple restrictions into one.
+ */
+public final class Composite implements RentalRestrictionExtension {
+
+  private final RentalRestrictionExtension[] exts;
+
+  private Composite(RentalRestrictionExtension... exts) {
+    for (var ext : exts) {
+      if (ext instanceof org.opentripplanner.service.vehiclerental.street.Composite) {
+        throw new IllegalArgumentException(
+          "Composite extension cannot be nested into one another."
```

Comment:
Not sure why you do not want this, and if not you could just flatten the structure preserving the order.

Reply:
I see that the implementation of the Composite becomes easier, but you could flatten it - that way it become more robust in about the same number of lines of code as now.

Reply:
I'm already flatten it in a static initializer `Composite.of`. This line is just a last safety check that I'm haven't made a mistake.